kernel/kernel-api/src/main/java/io/delta/kernel/internal/tablefeatures/TableFeatures.java

```diff
@@ -183,28 +183,44 @@ public boolean metadataRequiresFeatureToBeEnabled(Protocol protocol, Metadata me
     }
   }
 
-  public static final TableFeature VARIANT_RW_FEATURE = new VariantTypeTableFeature("variantType");
-  public static final TableFeature VARIANT_RW_PREVIEW_FEATURE =
-      new VariantTypeTableFeature("variantType-preview");
-
-  private static class VariantTypeTableFeature extends TableFeature.ReaderWriterFeature
-      implements FeatureAutoEnabledByMetadata {
-    VariantTypeTableFeature(String featureName) {
-      super(
-          /* featureName = */ featureName, /* minReaderVersion = */ 3, /* minWriterVersion = */ 7);
+  /* ---- Start: variantType ---- */
+  // Base class for variantType and variantType-preview features. Both features are same in terms
+  // of behavior and given the feature is graduated, we will enable the `variantType` by default
+  // if the metadata requirements are satisfied and the table doesn't already contain the
+  // `variantType-preview` feature.
+  private static class VariantTypeTableFeatureBase extends TableFeature.ReaderWriterFeature {
```

Comment:
For both of these, the preview feature is never auto-enabled-by-metadata right (needs to be manually added to the protocol)? Maybe note that I don't think it's immediately clear